WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java

```diff
@@ -378,9 +393,92 @@ public void onCreate(Bundle savedInstanceState) {
         // this fragment extends Support fragment.
         mViewModel = ViewModelProviders.of((FragmentActivity) getActivity(), mViewModelFactory)
                                        .get(ReaderPostListViewModel.class);
+
+        if (BuildConfig.INFORMATION_ARCHITECTURE_AVAILABLE && mIsMainReader) {
+            mViewModel.getCurrentSubFilter().observe(getActivity(), subfilterListItem -> {
+                if (isFollowing()) {
+                    setCurrentSubfilter(subfilterListItem, true);
+                }
+            });
+
+            mViewModel.getShouldShowSubFilters().observe(getActivity(), show -> {
+                mSubFilterComponent.setVisibility(show ? View.VISIBLE : View.GONE);
+            });
+        }
+
         mViewModel.start(mCurrentTag);
     }
 
+    private void setCurrentSubfilter(SubfilterListItem subfilterListItem, boolean requestNewerPosts) {
```

Comment:
Would it be possible to move this logic into the view model and only expose an endpoint with `LiveData<CurrentTagInfo>` and keep the object as a field in here?